### PR TITLE
[DISCUSS] JDBC compatibility docs

### PR DESCRIPTION
**Pull request by amotl**
Hi.

It's a very old patch, and it looks like we have most of the content already, but it may contain a few more interesting details.

With kind regards,
Andreas.

### CODE REVIEW

File: docs/index.txt [hunk 1]
```diff
@@ -72,8 +72,41 @@ The ``jdbc:`` prefix is optional. For example::
...
+* ``CallableStatement`` is not implemented as CRATE has no stored procedures to call
+
+For further details about compatibility with all possible JDBC features,
+see the ``ResultSet``, ``ResultSetMetaData`` and ``DatabaseMetaData`` implementations.
+
+Though only an optional feature, it is still worth mentioning:
+
+* the ``ResultSet`` implementation is read only (``TYPE_FORWARD_ONLY``, ``CONCUR_READ_ONLY``),
+  so changes to a ``ResultSet`` are not supported. Though DDL and DML statements are supported
+  using the ``Statement`` and ``PreparedStatement`` interfaces.
```
Comment (amotl) [inline docs/index.txt:95]: The first sentence is nothing new. The second one may add a slight bit of information.

File: docs/index.txt [hunk 2]
```diff
@@ -72,8 +72,41 @@ The ``jdbc:`` prefix is optional. For example::
...
+whose crate-client dependency matches the Crate server to connect to as good as possible.
+
+
 .. _`Bintray`: https://bintray.com/crate/crate/
 
 .. _`JCenter`: https://bintray.com/bintray/jcenter
 
 .. _`Squirrel`: http://squirrel-sql.sourceforge.net/
+
+.. _`JDBC 4.1`: http://download.oracle.com/otn-pub/jcp/jdbc-4_1-mrel-spec/jdbc4.1-fr-spec.pdf
```
Comment (amotl) [inline docs/index.txt:112]: Maybe?

File: docs/index.txt [hunk 3]
```diff
@@ -72,8 +72,41 @@ The ``jdbc:`` prefix is optional. For example::
     crate://localhost:4300
 
 
+Compatibility
+=============
+
+JDBC
+----
+
+This JDBC driver complies to the `JDBC 4.1`_ standard as good as possible and reasonable for
```
Comment (matriv) [inline docs/index.txt:81]: I'd rephrase that to:
`This JDBC driver complies to the `JDBC 4.1`_ standard to a very good level but not fully. The following mandatory features are not supported:`

File: docs/index.txt [hunk 4]
```diff
@@ -72,8 +72,41 @@ The ``jdbc:`` prefix is optional. For example::
...
+* ``DataSource`` is not implemented
+* ``CallableStatement`` is not implemented as CRATE has no stored procedures to call
+
+For further details about compatibility with all possible JDBC features,
+see the ``ResultSet``, ``ResultSetMetaData`` and ``DatabaseMetaData`` implementations.
+
+Though only an optional feature, it is still worth mentioning:
+
+* the ``ResultSet`` implementation is read only (``TYPE_FORWARD_ONLY``, ``CONCUR_READ_ONLY``),
+  so changes to a ``ResultSet`` are not supported. Though DDL and DML statements are supported
```
Comment (matriv) [inline docs/index.txt:94]: I would remove though:
```suggestion
  so changes to a ``ResultSet`` are not supported. DDL and DML statements are supported
```

File: docs/index.txt [hunk 5]
```diff
@@ -72,8 +72,41 @@ The ``jdbc:`` prefix is optional. For example::
...
+  using the ``Statement`` and ``PreparedStatement`` interfaces.
+
+Crate
+-----
+
+This JDBC driver can only be used with *Crate version 0.38.0* and higher.
+
+Besides using the most recent version, it is recommended to use the JDBC driver version
+whose crate-client dependency matches the Crate server to connect to as good as possible.
```
[the remark this answers is not in the document]
Comment (matriv) [reply docs/index.txt:103]: I agree, let's remote this and keep only the first sentence.

File: docs/index.txt (hunk 2 above)
Comment (matriv) [reply docs/index.txt:112]: yep, nice to have a link to the spec.

**Review by matriv — COMMENTED** [review]
Left some comments, thx!

**Comment by amotl** [comment]
Thanks for your feedback. I've converged it into GH-369.
